dist/test_tag.py

```diff
@@ -89,56 +104,67 @@
         # Check if line is valid
         if not line:
             # Check if invalid line after START_TAG
-            if is_start == True:
-                print("Error syntax in file " + filename)
-                exit()
+            if in_tag_block == True:
+                print("Syntax error in file: " + filename)
+                exit(1)
             else:
                 continue
 
         # Check if end of test tag
         if END_TAG in line:
             # END_TAG should not be before START_TAG
-            if is_start == False:
-                print("Error syntax in file " + filename + ". Unexpected tag: " + END_TAG)
-                exit()
+            if in_tag_block == False:
+                print("Syntax error in file: " + filename + ". Unexpected tag: " + END_TAG)
+                exit(1)
             # END_TAG should not be met before a test tag
             if is_file_ignored == False and is_file_tagged == False:
-                print("Error syntax in file " + filename + ". Missing test tag.")
-                exit()
-            is_end = True
+                print("Syntax error in file: " + filename + ". Missing test tag.")
+                exit(1)
             nb_valid_files = nb_valid_files + 1
             # Go to next file
             break
 
         # Check if start of test tag
         if START_TAG in line:
             # Only one START_TAG is allowed
-            if is_start == True:
-                print("Error syntax in file " + filename + ". Unexpected tag: " + START_TAG)
-                exit()
-            is_start = True
+            if in_tag_block == True:
+                print("Syntax error in file: " + filename + ". Unexpected tag: " + START_TAG)
+                exit(1)
+            in_tag_block = True
             continue
 
-        if is_start == True:
+        if in_tag_block == True:
+            tag = line
             # Check if file is ignored
             if is_file_ignored == True:
                 print("Unexpected value in ignored file: " + filename)
-                exit()
-            if line == IGNORE_FILE:
+                exit(1)
+            if tag == IGNORE_FILE:
                 nb_ignored_files = nb_ignored_files + 1
                 is_file_ignored = True
                 continue
-            # Check if current tag is valid
-            if not line in valid_tags:
-                print("Tag is not valid ! Add the new tag to test_tags.ok:\n" + line)
-                exit()
-            else:
-                is_file_tagged = True
+
+            # Validate the tag's correctness.
+            tag_count = validate_tag(tag, filename)
```

Comment:
`tag_count` does not seem used anymore. If we dont need it, we might as well return a bool from `validate_tag`

Reply:
bbdb81c9c